henryPuma/assignments/fastSlow/lc141/lc141.js

```diff
@@ -0,0 +1,84 @@
+// Problem Statement #
+
+// Given the head of a Singly LinkedList, write a function to determine if the LinkedList has a cycle in it or not.
+
+class Node {
+  constructor(value, next=null){
+    this.value = value;
+    this.next = next;
+  }
+}
+
+// Precondition: 
+// head: pointer to the start of the linked list
+// Postcondition: 
+// has_cycle returns a Boolean indicating whether or not the input linked list is a cycle
+
+/* General Idea: Have 2 pointers start at head (slow && fast)
+slow: iterates by 1 node,
+fast: iterates by 2 nodes,
+Eventually fast will traverse through the cycle faster and meet with slow at some arbitrary node,
+If fast doesn't meet slow and reaches null or the next pointer for fast is null then return fals,
+<=> we have reached the end of the linked list
+*/
+function has_cycle(head) {
+  // Set both pointers at head position
+  let slow = head, fast = head;
+  // While the pointer that can look ahead the farthest can move at least one forwards
+  while(fast && fast.next){
+    slow = slow.next; // Move by 1
+    fast = fast.next.next; // Move by 2
+    // At some point when both pointers meet then we definently have a cycle somewhere fast was able to re-encounter the slow pointer
+    if (fast === slow) return true; 
+  }
+  // Fast arrived at null, so there is no linked list 
+  return false;
+}
```

Comment:
This is an opinionated topic, but, many would agree that removing that inline comments from codebase before submitting is a good move. Moreover, make sure to type it twice!